src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -736,41 +679,6 @@ public override async Task<VersionStamp> GetTopLevelChangeTextVersionAsync(Cance
             return treeAndVersion.Version;
         }
 
-        public async Task<ImmutableDictionary<string, string>> GetAnalyzerOptionsAsync(string? projectFilePath, CancellationToken cancellationToken)
-        {
-            // We need to work out path to this document. Documents may not have a "real" file path if they're something created
-            // as a part of a code action, but haven't been written to disk yet.
-            string? effectiveFilePath = null;
-
-            if (FilePath != null)
-            {
-                effectiveFilePath = FilePath;
-            }
-            else if (Name != null && projectFilePath != null)
-            {
-                var projectPath = PathUtilities.GetDirectoryName(projectFilePath);
-
-                if (!RoslynString.IsNullOrEmpty(projectPath) &&
-                    PathUtilities.GetDirectoryName(projectFilePath) is string directory)
-                {
-                    effectiveFilePath = PathUtilities.CombinePathsUnchecked(directory, Name);
-                }
-            }
-
-            if (effectiveFilePath != null)
-            {
-                var analyzerConfigSet = await _analyzerConfigSetSource.GetValueAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Was this code moved to Document because it no longer requires `_analyzerConfigSetSource`?

Reply:
@agocke?

Reply:
Yes, but the broader reason is that analyzer config options are no longer a "document-level" setting. The design is to query the project for analyzer config settings for a particular document, not to query a document in isolation. That's a major theme of this refactoring in both the compiler and IDE.